### ISSUE TITLE
Thread purging

### ISSUE THREAD

**Issue by bakape** [bakape/meguca#1061]
- [ ] When purging a thread, purge all of it's posts too
- [ ] Purged posts should style just like deleted posts - need to modify the `isDeleted()` methods.

**Comment by Chiiruno** [comment]
Should this be for the entire thread, or just per-IP?
There would be no point in purging an entire thread if it is per-IP, since normal users should be allowed to post in it without being purged also.
I suppose if someone were to post something illegal, they could also do it on different IPs each time, in that thread. So, purging the entire thread scorched Earth style would make sense.
I suspect this might fit into the anti-spam category, assuming the purpose of this is to eliminate threads where someone posts something illegal each time in it with a different IP each time.
In that case, OP should be recursively purged by-IP, for all posts on the site, and the rest of the thread should be purged only within that thread.

**Comment by Chiiruno** [comment]
Also an optional permanent ban checkbox should be added for purging posts.

**Comment by bakape** [comment]
Spammers favour to create new threads instead of posts. Probably because
meguca moves too fast and their posts would be drowned quick.

**Comment by Chiiruno** [comment]
Yeah, per-IP would be pointless, since a spammer or someone posting something illegal may just use different IPs each request, best purge the entire thread, and only apply the IP-wide purge to the OP.
Optional permanent ban checkbox should still be added though I think.
